dspace-api/src/main/java/org/dspace/scripts/DSpaceCommandLineParameter.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.scripts;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * This class serves as a representation of a command line parameter by holding a String name and a String value
+ */
+public class DSpaceCommandLineParameter {
+    private String name;
+    private String value;
+
+    public static String SEPARATOR = "|||";
+
+    /**
+     * This constructor will take a String key and String value and store them in their appriopriate fields
+     * @param key   The String value to be stored as the name of the parameter
+     * @param value The String value to be stored as the value of the parameter
+     */
+    public DSpaceCommandLineParameter(String key, String value) {
+        this.name = key;
+
+        if (StringUtils.isBlank(value)) {
+            this.value = null;
+        } else {
+            this.value = value;
+        }
+    }
+
+    /**
+     * This constructors accepts a single parameter String that is defined as e.g. "-c test" and it'll parse this
+     * String into the key "-c" and value "test" to then call the other constructor with those parameters
+     * @param parameter The String parameter
+     */
+    public DSpaceCommandLineParameter(String parameter) {
+        this(StringUtils.substringBefore(parameter, " "), StringUtils.substringAfter(parameter, " "));
```

Comment:
This logic seems potentially fragile.  What if someone accidentally types in *two spaces*, e.g. `-c[space][space]test`.

Also, have you investigated whether there is a way to potentially use the tools/API in Apache Commons CLI to do this parameter parsing for us (instead of doing string parsing)?  If we could just use that CLI API directly, it'd be better than rewriting it ourselves.

Reply:
This constructor is currently only used in an environment from which its input comes directly from the Database. It wasn't being exposed to user input as that would have been processed and committed into the database before this code was ever called. However this is a very valid point and a risk to be fragile as you rightfully noted, which is why we opted to make this constructor protected to not further risk any mistakes to be made by leaving this constructor as public.
We've moved the Process (and all its related classes) to the org.dspace.scripts package and made that constructor Protected so it can only be used from in there.
As for the Apache Commons libraries, we have indeed looked through them but we had not found a library that seemed useful to use. This is why we opted to do the parsing ourselves